pkg/cloud/services/gc/loadbalancer.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gc
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+)
+
+func (s *Service) deleteLoadBalancers(ctx context.Context, resources []*AWSResource) error {
+	for _, resource := range resources {
+		if !s.isLoadBalancerToDelete(resource) {
+			s.scope.V(4).Info("Resource not a load balancer for deletion", "arn", resource.ARN.String())
+			continue
+		}
+
+		if strings.HasPrefix(resource.ARN.Resource, "loadbalancer/app/") {
+			s.scope.V(5).Info("Deleting ALB for Service", "arn", resource.ARN.String())
+			if err := s.deleteLoadBalancerV2(ctx, resource.ARN.String()); err != nil {
+				return fmt.Errorf("deleting ALB: %w", err)
+			}
+			continue
+		}
+		if strings.HasPrefix(resource.ARN.Resource, "loadbalancer/net/") {
+			s.scope.V(5).Info("Deleting NLB for Service", "arn", resource.ARN.String())
+			if err := s.deleteLoadBalancerV2(ctx, resource.ARN.String()); err != nil {
+				return fmt.Errorf("deleting NLB: %w", err)
+			}
+			continue
+		}
+		if strings.HasPrefix(resource.ARN.Resource, "loadbalancer/") {
+			name := strings.ReplaceAll(resource.ARN.Resource, "loadbalancer/", "")
+			s.scope.V(5).Info("Deleting classic ELB for Service", "arn", resource.ARN.String(), "name", name)
+			if err := s.deleteLoadBalancer(ctx, name); err != nil {
+				return fmt.Errorf("deleting classic ELB: %w", err)
+			}
+			continue
+		}
```

Comment:
So, something like:

```go
		switch {
		case strings.HasPrefix(resource.ARN.Resource, "loadbalancer/app/"),
			strings.HasPrefix(resource.ARN.Resource, "loadbalancer/net/"):
			s.scope.V(5).Info("Deleting ALB/NLB for Service", "arn", resource.ARN.String())
			if err := s.deleteLoadBalancerV2(ctx, resource.ARN.String()); err != nil {
				return fmt.Errorf("deleting ALB/NLB: %w", err)
			}
		case strings.HasPrefix(resource.ARN.Resource, "loadbalancer/"):
			name := strings.ReplaceAll(resource.ARN.Resource, "loadbalancer/", "")
			s.scope.V(5).Info("Deleting classic ELB for Service", "arn", resource.ARN.String(), "name", name)
			if err := s.deleteLoadBalancer(ctx, name); err != nil {
				return fmt.Errorf("deleting classic ELB: %w", err)
			}
		}
```

Reply:
With the suggested isLBv2 function the first case could be further simplified. And then isLBv2 later can be used during deletion to determine which input value to set.

Reply:
Changed to a switch. But I've kept them as separate cases in the switch so that we get different log messages.

Reply:
>Changed to a switch. But I've kept them as separate cases in the switch so that we get different log messages.

![understandable-have-a-nice-day-1](https://user-images.githubusercontent.com/182850/181564511-050eebe3-9b43-47d0-9e91-737dc84c72cf.jpeg)

Reply:
:rofl: